### PR TITLE
Fix(mongo): verify / resetPassword use writeResult

### PR DESCRIPTION
**Pull request by sdwvit**
### Description

Consider default verify / change password templates for mongo. Let's add console.log before each of the callback executions to log `count > 0` check result into console, and hit 'Save & Try'.

Script for verify:
```
   users.update(query, { $set: { email_verified: true } }, function (err, count) {
      client.close();
        console.log(err, count > 0);
      if (err) return callback(err);
      callback(null, count > 0);
    });
```

Produces:

Screen Shot 2021-11-19 at 1 18 49 PM

🤔 ❓ ❓ ❓

Script for change password:

```
   users.update({ email: email }, { $set: { password: hash } }, function (err, count) {
        client.close();
        console.log(err, count > 0);
        if (err) return callback(err);
        callback(null, count > 0);
      });
```

Produces:

Screen Shot 2021-11-19 at 1 17 54 PM

As we can see, `count > 0` results in `false`, however manhattan happily tells us everything works as it should. This is not true 🥶. 
If we change console.log to dump the whole `count` variable, we see:

Screen Shot 2021-11-19 at 1 19 21 PM

### Explanation

Mongo doc tells us, that update() call returns WriteResult object, with the result.nModified property, which is meant to be used here most likely.
https://docs.mongodb.com/manual/reference/method/db.collection.update/#std-label-writeresults-update
But it's also the `Try` script which works incorrectly. Luckily, real auth0-users code executes and checks results correctly and throws an error, if second callback parameter is false in these scripts.

### Fix
One part is this fix for templates. Other part should happen in code which implements the `Try` and `Save & Try` functionality.

### CODE REVIEW

File: src/scripts/mongo/change_password.js [hunk 1]
```diff
@@ -15,10 +15,10 @@ function changePassword(email, newPassword, callback) {
         return callback(err);
       }
 
-      users.update({ email: email }, { $set: { password: hash } }, function (err, count) {
+      users.update({ email: email }, { $set: { password: hash } }, function (err, writeResult) {
         client.close();
         if (err) return callback(err);
-        callback(null, count > 0);
+        callback(null, writeResult.result.nModified > 0);
```
Comment (hernanhht) [inline src/scripts/mongo/change_password.js:21]: Doing the whole `writeResult.result.nModified` chain call seems a bit dangerous given that we don't know if there is a condition where mongo might not return some of those objects producing a TypeError. Maybe we can do this in a safer way, something like:
```js
const modified = writeResult && writeResult.result && writeResult.result.nModified;
callback(null, writeResult.result.nModified > 0);
```

File: src/scripts/mongo/verify.js [hunk 2]
```diff
@@ -9,11 +9,11 @@ function verify (email, callback) {
     const users = db.collection('users');
     const query = { email: email, email_verified: false };
 
-    users.update(query, { $set: { email_verified: true } }, function (err, count) {
+    users.update(query, { $set: { email_verified: true } }, function (err, writeResult) {
       client.close();
 
       if (err) return callback(err);
-      callback(null, count > 0);
+      callback(null, writeResult.result.nModified > 0);
```
Comment (hernanhht) [inline src/scripts/mongo/verify.js:16]: Same thing here

**Review by hernanhht — COMMENTED** [review]
I'm wondering if the original script used to work fine in an older version of mongo

**Comment by sdwvit** [comment]
Right now mongo version used for these script always returns WriteResult object, or non-null error as a first parameter of callback. Good idea to check for nullity, as we don't guaranty mongo api doesn't change in future.

File: src/scripts/mongo/change_password.js (hunk 1 above)
Comment (sdwvit) [reply src/scripts/mongo/change_password.js:21]: what should happen if modified is null | undefined?

**Comment by sdwvit** [comment]
Issue is present in other templates for mongo, need to fix those in this pr as well

**Pull request closed without merging**